oktaawscli/okta_auth.py

```diff
@@ -26,25 +25,22 @@ def __init__(self, okta_profile, verbose, logger,
         self.verbose = verbose
         self.okta_auth_config = okta_auth_config
         self.https_base_url = "https://%s" % okta_auth_config.base_url_for(okta_profile)
-        self.username = okta_auth_config.username_for(okta_profile)
-        self.password = okta_auth_config.password_for(okta_profile)
         self.factor = okta_auth_config.factor_for(okta_profile)
         self.app = okta_auth_config.app_for(okta_profile)
 
         okta_info = os.path.join(os.path.expanduser('~'), '.okta-token')
         if not os.path.isfile(okta_info):
             open(okta_info, 'a').close()
 
-
     def primary_auth(self):
         """ Performs primary auth against Okta """
         # check if token is valid
         session_id = self.get_cached_session_id()
         if session_id is not None:
             return session_id
         auth_data = {
-            "username": self.username,
-            "password": self.password
+            "username": self.okta_auth_config.username_for(self.okta_profile),
+            "password": self.okta_auth_config.password_for(self.okta_profile)
```

Comment:
By moving these method calls here we now have Okta Auth first check if the Okta session from the cached token is valid before asking the user for their username and password.